src/vs/workbench/contrib/positronIPyWidgets/browser/positronIPyWidgetsService.ts

```diff
@@ -117,15 +133,18 @@ export class PositronIPyWidgetsService extends Disposable implements IPositronIP
 			// Fire the onDidCreatePlot event.
 			const client = disposables.add(new NotebookOutputPlotClient(webview, message));
 			this._onDidCreatePlot.fire(client);
-		}));
+		};
+
+		disposables.add(session.onDidReceiveRuntimeMessageResult(handleMessageOutput));
+		disposables.add(session.onDidReceiveRuntimeMessageOutput(handleMessageOutput));
 
 		// Dispose when the session ends.
 		disposables.add(session.onDidEndSession((e) => {
 			disposables.dispose();
```

Comment:
When we dispose, shouldn't we also delete the entry from `SessionToDisposablesMap`? Could be handled as described in my previous comment.